lib/nationbuilder/utils.rb

```diff
@@ -0,0 +1,21 @@
+module NationBuilder::Utils
+  module QueryParams
+
+    def self.encode(value, key = nil, output = {})
```

Comment:
Depending on the case, we do use it on `l6` and `l7` when rebuilding the output - the recursive aspect handles nested params. 

I imagined existing specs would catch things if they went sideways here, it wouldn't hurt to have this better tested as well though (and would aid in refactoring this out of a class method) - I'll dig into that more

Reply:
yeah after taking a second look I think it makes sense as a recursive function instead of an object.  However as it's hard to reason about, I think it'd be useful to have some complex examples implemented as unit tests that show how the encoding works